Puyolib/DebugLog.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2023, LossRain
+// SPDX-License-Identifier: GPLv3
+
+#pragma once
+
+#include <string>
+#include <deque>
+#include <iostream>
+#include <functional>
+
+namespace ppvs {
+
+enum class DebugMessageType {
+	NONE = 0,
+	ERROR = 1,
+	WARNING = 2,
+	INFO = 4,
+	DEBUG = 8
+};
+
+/* A more precise debug message class, might be used in exception handling and other scenarios
+ * where we don't want to handle raw strings.
+ * */
+class DebugMessage {
+public:
+	DebugMessage(std::basic_string<char> text, DebugMessageType severity = DebugMessageType::NONE) : m_text(text), m_severity(severity) {};
```

Comment:
Another instance of `std::basic_string<char>` that should most likely just be `std::string`